tests/testserver/server.py

```diff
@@ -39,6 +39,10 @@ def __init__(self, handler, host='localhost', port=0, requests_to_handle=1, wait
         self.ready_event = threading.Event()
         self.stop_event = threading.Event()
 
+        self.server_sock = self._create_socket_and_bind()
```

Comment:
I gave an answer; it was just a rather unsatisfying one about setting instance variables in the constructor. :smile:

I think I was not seeing the synchronization between `__enter__` and `run` earlier, which made me want to set this before starting the server thread. I'll go ahead and move this back to `run`.